### PR TITLE
fix: simplify tx history cleanup and fixes

### PR DESCRIPTION
**Pull request by kaladinlight**
## Description

Some additional tx history cleanup (simple view), see screenshots for visual changes.

- Adds stricter spam asset filters (so many shit nfts, oh wait, that is all nfts... :P)
- `TransactionGenericRow` handles all common transfer types
- Rename `UnknownTransaction` -> `TransactionCommon`
- Remove `TransactionSend` and `TransactionReceive` as these are currently act the same as `TransactionCommon`

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

Low - display only

> What protocols, transaction types or contract interactions might be affected by this PR?

Affects all transaction types

## Testing

- Ensure transaction history is still displayed accurately

## Screenshots (if applicable)

|Before|After|
|----------|-------|
|image|image|
|image|image|
|image|image|
|image|image|
|image|image|
|image|image|
|image|image|
|image|image|
|image|image|

### CODE REVIEW

**Review by 0xApotheosis — APPROVED** [review]
✅ **We now get the correct sign for deposit transactions (this diff left, develop right):**

Screenshot 2024-02-20 at 11 00 38 am

✅ **Spam NFTs correctly filtered out of the transaction list:**

This diff:
Screenshot 2024-02-20 at 11 03 04 am

Develop:

Screenshot 2024-02-20 at 11 04 01 am

❓ The behavior of the relative headers has changed, e.g. this transaction used to show as "This week", though I believe that's because we used to use "the last 7 days" as that heuristic, whereas now we use calendar weeks. The same behavior occurs with "This month". This might be intentional, in that case this is fine 🙏 

Screenshot 2024-02-20 at 11 05 33 am

✅ All other transactions that I checked matched production perfectly.

File: src/components/AssetIconWithBadge.tsx [hunk 1]
```diff
@@ -61,7 +61,7 @@ export const AssetIconWithBadge: React.FC<AssetIconWithBadgeProps> = ({
       }
       if (transfers.length > 2 && compareIndex === 3) return null
       return (
-        <>
+        <Box key={index}>
```
Comment (gomesalexandre) [inline src/components/AssetIconWithBadge.tsx:64]: Less spew 🧠

**Review by gomesalexandre — COMMENTED** [review]
Diff looking good according to the PR description changeset, conceptual stamp pending runtime testing

**Review by gomesalexandre — APPROVED** [review]
Tested against prod (right):

- Confirmed we're now less spammy on the NFT side of things ✅ 

image

- UTXO Txs aren't showed as "2 assets" ✅ 

image

- "Unknown" (common) transactions are now showing the native asset/chain, and date ✅ 

image

- Add Liquidity is now more descriptive of the underlying assets ✅ 

image

- Unstake is now more descriptive of the underlying assets ✅ 

image

- Common transaction is more descriptive WRT the underlying assets ✅ 

image

**Comment by kaladinlight** [comment]
@0xApotheosis 
> The behavior of the relative headers has changed, e.g. this transaction used to show as "This week", though I believe that's because we used to use "the last 7 days" as that heuristic, whereas now we use calendar weeks. The same behavior occurs with "This month". This might be intentional, in that case this is fine 🙏

This was an intentional update upon validating with @reallybeard the desired behavior.

File: src/components/TransactionHistoryRows/TransactionGenericRow.tsx [hunk 2]
```diff
@@ -93,49 +134,141 @@ export const TransactionGenericRow = ({
         />
       )
     }
-  }, [hasManySends, hasSendAndReceive, transfersByType.Send])
 
+    if (hasManySendAssets) {
+      const assets = transfersByType.Send.map(transfer => transfer.asset.symbol)
```
[the remark this answers is not in the document]
Comment (kaladinlight) [reply src/components/TransactionHistoryRows/TransactionGenericRow.tsx]: https://github.com/shapeshift/web/pull/6231/commits/9530cba9e81b9e9234ee5825a8b6c7793c73e446

**Pull request merged**